tajo-core/src/main/java/org/apache/tajo/querymaster/Repartitioner.java

```diff
@@ -648,10 +644,15 @@ public static void scheduleRangeShuffledFetches(TaskSchedulerContext schedulerCo
         throw new IOException("Can't get table meta data from catalog: " +
             PlannerUtil.getStoreTableName(masterPlan.getLogicalPlan()));
       }
-      ranges = TableSpaceManager.getStorageManager(stage.getContext().getConf(), storeType)
-          .getInsertSortRanges(stage.getContext().getQueryContext(), tableDesc,
-              sortNode.getInSchema(), sortSpecs,
-              mergedRange);
+
+      Tablespace space = TablespaceManager.getAnyByScheme(storeType).get();
+      ranges = space.getInsertSortRanges(
```

Comment:
In original implementation, it guarantees that TablespaceManager.getAnyByScheme(storeType) will always return the proper tablespace handler if ``sortNode.getSortPurpose()`` is ``STORAGE_SPECIFIED``.

But, its implementation is not intuitive. We should improve them. But, it requires lots of extra changes. I'll continue to improve storage layer APIs. It will be improved later too.